### PR TITLE
Add releasing doc

### PR DESCRIPTION
**Pull request by cijothomas**
Listed the steps I followed to do the release to nuget.

Note: This entire process will be automated, this PR is just to make sure the steps are documented somewhere public, so that anyone can write the automation now.

### CODE REVIEW

File: Releasing.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Release process.
+
+Only for Maintainers.
+
+1. Tag with version to be released eg: git tag -a 0.4.0-beta -m ".4.0.0-beta"
```
Comment (alanwest) [inline Releasing.md:5]: ".4.0.0-beta" -> "0.4.0-beta"

File: Releasing.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+   git push origin 0.4.0-beta
+
+2. Submit a PR and merge it. (any harmless PR) To trigger CI for master. (no
+   longer required, but confirm)
+
+3. https://github.com/open-telemetry/opentelemetry-dotnet/actions?query=workflow%3A%22Pack+and+publish+to+Myget%22
+   Wait for the above CI build pipeline to finish. Its triggered when the above
```
Comment (alanwest) [inline Releasing.md:12]: Its triggered -> It is triggered

File: Releasing.md (hunk 1 above)
Comment (eddynaka) [reply Releasing.md:5]: and, i would try to add/test the `--sign` or `-s` to generate a verified tag. (that doesn't change a thing AFAIK)

File: Releasing.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+Release process.
```
Comment (SergeyKanzhelev) [inline Releasing.md:1]:
```suggestion
# Release process.
```

File: Releasing.md (hunk 1 above)
Comment (SergeyKanzhelev) [reply Releasing.md:5]:
```suggestion
1. Tag with version to be released eg: git tag -a 0.4.0-beta -m "0.4.0-beta"
```

File: Releasing.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+   longer required, but confirm)
+
+3. https://github.com/open-telemetry/opentelemetry-dotnet/actions?query=workflow%3A%22Pack+and+publish+to+Myget%22
+   Wait for the above CI build pipeline to finish. Its triggered when the above
+   PR is merged. At the end of this, myget will have the packages.
+
+4. Validate using myget packages. Basic sanity checks :)
+
+5. From the above build, get the artifacts from the drop, which has all the
+   nuget packages.
```
Comment (SergeyKanzhelev) [inline Releasing.md:18]:
```suggestion
   nuget packages and symbols (*.snupkg files).
```

File: Releasing.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+6. Copy the nuget into a local folder.
+
+7. Download latest nuget.exe to this folder: https://www.nuget.org/downloads
+
+8. Obtain the API key from nuget.org (Only maintainers have access)
+
+9. run the following command from powershell from the above folder..
```
Comment (SergeyKanzhelev) [inline Releasing.md:26]:
```suggestion
9. Run the following command from powershell from the above folder..
```

File: Releasing.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
```
Comment (SergeyKanzhelev) [inline Releasing.md:27]:
```suggestion
10. Remove API key so it wouldn't be re-used unintentionally
```

File: Releasing.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+3. https://github.com/open-telemetry/opentelemetry-dotnet/actions?query=workflow%3A%22Pack+and+publish+to+Myget%22
+   Wait for the above CI build pipeline to finish. Its triggered when the above
+   PR is merged. At the end of this, myget will have the packages.
+
+4. Validate using myget packages. Basic sanity checks :)
+
+5. From the above build, get the artifacts from the drop, which has all the
+   nuget packages.
+
+6. Copy the nuget into a local folder.
```
Comment (SergeyKanzhelev) [inline Releasing.md:20]:
```suggestion
6. Copy all the nuget files and symbols into a local folder.
```

File: Releasing.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+Release process.
+
+Only for Maintainers.
+
+1. Tag with version to be released eg: git tag -a 0.4.0-beta -m ".4.0.0-beta"
+   git push origin 0.4.0-beta
+
+2. Submit a PR and merge it. (any harmless PR) To trigger CI for master. (no
```
Comment (SergeyKanzhelev) [inline Releasing.md:8]: hm... can we just configure CI to build from tags?

Comment (eddynaka) [reply Releasing.md:8]: this section will be removed. We already did this part using the CI :)

File: OpenTelemetry.sln [hunk 9]
```diff
@@ -16,6 +16,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 		NuGet.config = NuGet.config
 		OpenTelemetry.proj = OpenTelemetry.proj
 		README.md = README.md
+		Releasing.md = Releasing.md
```
Comment (reyang) [inline OpenTelemetry.sln:19]: Consider making this uppercase:
1. align with other top level *.md files (README, CONTRIBUTING)
2. align with OpenTelemetry Python https://github.com/open-telemetry/opentelemetry-python/blob/master/RELEASING.md

File: Releasing.md (hunk 3 above)
Comment (reyang) [reply Releasing.md:1]: And remove the period

File: Releasing.md (hunk 5 above)
Comment (reyang) [reply Releasing.md:26]: powershell -> PowerShell https://en.wikipedia.org/wiki/PowerShell

File: Releasing.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+6. Copy the nuget into a local folder.
+
+7. Download latest nuget.exe to this folder: https://www.nuget.org/downloads
+
+8. Obtain the API key from nuget.org (Only maintainers have access)
+
+9. run the following command from powershell from the above folder..
+
+.\nuget.exe setApiKey <your_API_key> get-childitem | where {$_.extension -eq
```
Comment (reyang) [inline Releasing.md:28]: make this a code block

File: Releasing.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+   PR is merged. At the end of this, myget will have the packages.
+
+4. Validate using myget packages. Basic sanity checks :)
+
+5. From the above build, get the artifacts from the drop, which has all the
+   nuget packages.
+
+6. Copy the nuget into a local folder.
+
+7. Download latest nuget.exe to this folder: https://www.nuget.org/downloads
```
Comment (reyang) [inline Releasing.md:22]: latest -> the latest

Comment (reyang) [reply Releasing.md:22]: to which folder?

File: Releasing.md (hunk 1 above)
Comment (reyang) [inline Releasing.md:5]: nit: eg -> e.g.

File: Releasing.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+   git push origin 0.4.0-beta
+
+2. Submit a PR and merge it. (any harmless PR) To trigger CI for master. (no
+   longer required, but confirm)
+
+3. https://github.com/open-telemetry/opentelemetry-dotnet/actions?query=workflow%3A%22Pack+and+publish+to+Myget%22
+   Wait for the above CI build pipeline to finish. Its triggered when the above
+   PR is merged. At the end of this, myget will have the packages.
+
+4. Validate using myget packages. Basic sanity checks :)
```
Comment (reyang) [inline Releasing.md:15]: nit: myget -> MyGet https://www.myget.org

File: Releasing.md (hunk 4 above)
Comment (reyang) [inline Releasing.md:18]: nit: nuget -> NuGet https://www.nuget.org

File: Releasing.md (hunk 3 above)
Comment (eddynaka) [reply Releasing.md]: updated!

File: Releasing.md (hunk 1 above)
Comment (eddynaka) [reply Releasing.md]: updated!

Comment (eddynaka) [reply Releasing.md]: updated!

File: Releasing.md (hunk 8 above)
Comment (eddynaka) [reply Releasing.md]: just removed this section.

File: Releasing.md (hunk 2 above)
Comment (eddynaka) [reply Releasing.md]: updated!

File: Releasing.md (hunk 12 above)
Comment (eddynaka) [reply Releasing.md]: updated!

File: Releasing.md (hunk 4 above)
Comment (eddynaka) [reply Releasing.md]: updated!

Comment (eddynaka) [reply Releasing.md]: updated!

File: Releasing.md (hunk 7 above)
Comment (eddynaka) [reply Releasing.md]: updated!

File: Releasing.md (hunk 11 above)
Comment (eddynaka) [reply Releasing.md]: changed the phrase so we won't mention a folder

File: Releasing.md (hunk 5 above)
Comment (eddynaka) [reply Releasing.md]: updated!

File: Releasing.md (hunk 10 above)
Comment (eddynaka) [reply Releasing.md:27]: updated!

File: Releasing.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+   git tag -a 0.4.0-beta -m "0.4.0-beta"
+   git push origin 0.4.0-beta
+   ```
+
+2. Wait for the [Pack and publish to MyGet
+   workflow](https://github.com/open-telemetry/opentelemetry-dotnet/actions?query=workflow%3A%22Pack+and+publish+to+Myget%22)
+   to finish. It's triggered when the you create a new tag. At the end of this,
+   MyGet will have the packages.
+
+3. Validate using MyGet packages. Basic sanity checks :)
```
Comment (ThomsonTan) [inline Releasing.md:17]: List the commands (like nuget) to do basic sanity check?

File: Releasing.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+4. From the above build, get the artifacts from the drop, which has all the
+   nuget packages and symbols (*.snupkg files).
+
+5. Copy all the nuget files and symbols into a local folder.
```
Comment (ThomsonTan) [inline Releasing.md:22]: Is it necessary to copy symbols to local folder?

Comment (eddynaka) [reply Releasing.md:22]: If we don't do that, it won't publish automatically to nuget

File: Releasing.md (hunk 13 above)
Comment (cijothomas) [reply Releasing.md]: dont plan to write this, as we intend to have this entire thing automated with no human touch.

File: RELEASING.md [hunk 15]
```diff
@@ -0,0 +1,99 @@
+# Release process
+
+Only for Maintainers.
+
+1.Tag with version to be released e.g.:
```
Comment (SergeyKanzhelev) [inline RELEASING.md:5]: is space lacking intentionally? If list got restarted with the spaces - you can either remove newlines between item and code example or just change to the checkbox style 

``` markdown
- [ ] Blah
```

It should be rendered to the chackbox

File: RELEASING.md [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+
+```powershell
+    $changelogs = Get-ChildItem -Path . -Recurse -Filter changelog.md
+    foreach ($changelog in $changelogs)
+    {
+     (Get-Content -Path $changelog.FullName) -replace "Unreleased", "Unreleased
+
+    ## 0.6.0-beta.1
+
+    Released 2020-Sep-15" | Set-Content -Path $changelog.FullName
```
Comment (SergeyKanzhelev) [inline RELEASING.md:62]: the date can probably be scripted in powershell

File: RELEASING.md [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+   The actual version would be the tag name from step1 appended with
+   ".1"
+
+```powershell
+    $changelogs = Get-ChildItem -Path . -Recurse -Filter changelog.md
+    foreach ($changelog in $changelogs)
+    {
+     (Get-Content -Path $changelog.FullName) -replace "Unreleased", "Unreleased
+
+    ## 0.6.0-beta.1
```
Comment (SergeyKanzhelev) [inline RELEASING.md:60]: theoretically you can take it from the current tag on the branch. Not sure if it worth scripting it

**Review by SergeyKanzhelev — APPROVED** [review]

**Pull request merged**